src/main/resources/static/js/mods.js

```diff
@@ -27,6 +27,19 @@ angular.module('Mccy.mods',[
             $scope.compatibleServerTypes = $scope.mod.serverTypes.map(function(envType){
                 return $filter('serverTypeLabel')(envType);
             }).join(', ');
+
+            $scope.applicableVersions = Versions.query({type:$scope.mod.serverTypes[0]});
```

Comment:
This and the `$scope.update` usage are redundant. I like your idea of lazy loading the versions until they actually click to edit a mod's version. So, just get rid of this pre-load of `$scope.applicableVersions`.